src/mds/Locker.cc

```diff
@@ -3594,28 +3595,27 @@ bool Locker::any_late_revoking_caps(xlist<Capability*> const &revoking) const
     } else {
       utime_t now = ceph_clock_now();
       utime_t age = now - (*p)->get_last_revoke_stamp();
-      if (age <= mds->mdsmap->get_session_timeout()) {
+      if (age <= timeout) {
           return false;
       } else {
           return true;
       }
     }
 }
 
-
-void Locker::get_late_revoking_clients(std::list<client_t> *result) const
+void Locker::get_late_revoking_clients(std::list<client_t> *result,
+                                       double timeout) const
 {
-  if (!any_late_revoking_caps(revoking_caps)) {
+  if (!any_late_revoking_caps(revoking_caps, timeout)) {
     // Fast path: no misbehaving clients, execute in O(1)
     return;
   }
 
   // Slow path: execute in O(N_clients)
-  std::map<client_t, xlist<Capability*> >::const_iterator client_rc_iter;
-  for (client_rc_iter = revoking_caps_by_client.begin();
+  for (auto client_rc_iter = revoking_caps_by_client.begin();
```

Comment:
this is cosmetic right? In that case I'd prefer `for (auto& p : revoking_caps_by_client)`

Reply:
sure